include/deal.II/lac/linear_operator.h

```diff
@@ -1118,6 +1371,227 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
+
+    Range_1 &tmp_f1 = *(vector_memory_f.alloc());
+    Range_1 &tmp_f2 = *(vector_memory_f.alloc());
+
+    // Reinitialise in context of how they'll be used
+    B.reinit_range_vector(tmp_f1, /*bool omit_zeroing_entries =*/ true);
+    C.reinit_range_vector(tmp_f2, /*bool omit_zeroing_entries =*/ true);
+
+    // Solve for second field
+    // Need to form x such that x = A_inv*(f - B*y)
+    B.vmult(tmp_f1, y); // tmp1 = B*y
+    tmp_f2 = f;
+    tmp_f2 -= tmp_f1; // tmp2 = f - B*y
+    try
+      {
+        A_inv.vmult(x, tmp_f2); // x = A_inv*(f-B*y)
+      }
+    catch (...)
+      {
+        AssertThrow(false,
+                    ExcMessage("No convergence in A_inv vmult operation"));
+      }
+
+    // Need to reverse g*-->g performed in preprocessing
+    // step so that accurate residual quantification and
+    // other local static condensation operations, based
+    // on g, can be computed after solution process.
+    // i.e. Need to return to g = g* + C*A_inv*f
+    if (reverse_condensation == true)
+      preprocess(A_inv,C,
+                 f,g,
+                 /*forward_op =*/ false);
+  }
+
+  /**
+   * @relates LinearOperator
+   *
+   * Solve the general system of equations
+   * @f{eqnarray*}
+    \mathbf{K}\mathbf{d} = \mathbf{h}
+    \quad \Rightarrow\quad
+    \left(\begin{array}{cc}
+      A & B \\ C & D
+    \end{array}\right)
+    \left(\begin{array}{cc}
+      x \\ y
+    \end{array}\right)
+    =
+    \left(\begin{array}{cc}
+      f \\ g
+    \end{array}\right),
+   * @f}
+   * using the Schur complement.
+   *
+   * @note This requires the pre-construction of @p A_inv,
+   * the inverse operation of the (square) matrix @p A.
+   * The @p solver and @p preconditioner are used to construct
+   * the operation representing the inverse of the Schur
+   * complement, and should thus be chosen accordingly.
+   *
+   * @note The preconditioner could also be an approximation to the
+   * inverse of Schur complement that is constructed, naturally, using
+   * the function schur_complement().
+   *
+   * @see
+   * @ref GlossBlockLA "Block (linear algebra)"
+   * @author Jean-Paul Pelteret, 2015
+   *
+   * @ingroup LAOperators
+   */
+  template <typename Range_1, typename Domain_1,
+            typename Range_2, typename Domain_2,
+            typename Solver, typename Preconditioner>
+  void
+  solve_system (const LinearOperator<Range_1, Domain_1> &A_inv,
+                const LinearOperator<Range_1, Domain_2> &B,
+                const LinearOperator<Range_2, Domain_1> &C,
+                const LinearOperator<Range_2, Domain_2> &D,
+                Range_1                                 &x,
+                Range_2                                 &y,
+                const Range_1                           &f,
+                const Range_2                           &g,
+                Solver                                  &solver,
+                const Preconditioner                    &preconditioner)
```

Comment:
I think that refactoring the functions above obsoletes this wrapper [1].
The resulting user interface would look like
```
auto S = schur_complement(A_inv, B, C, D);
auto S_inv = inverse_operator(S, solver, preconditioner);

auto rhs = preprocess(A_inv, C, f, g) // PackagedOperation
y = S_inv * rhs;
x = postprocess(A_inv, B, C, y, f, g);
```

[1] ... and it contradicts the very idea behind ```LinearOperator``` to get rid of those one purpose high level interfaces...